File: app/src/main/java/com/rahmitufanoglu/retrofitdemo/NetworkChangeReceiver.java
package com.rahmitufanoglu.retrofitdemo;


import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.support.v7.app.AlertDialog;
import android.widget.Toast;


public class NetworkChangeReceiver extends BroadcastReceiver {

    private Context mContext;

    @Override
    public void onReceive(Context context, Intent intent) {
        mContext = context;

        ConnectivityManager connectivityManager = (ConnectivityManager) mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
        boolean isConnected = networkInfo != null && networkInfo.isConnected();
        if (isConnected) {
            Toast.makeText(mContext, "Internet connected", Toast.LENGTH_LONG).show();
            mContext.startActivity(intent);
            ((Activity) mContext).finish();
        } else {
            Toast.makeText(mContext, "Internet connection lost", Toast.LENGTH_LONG).show();
            checkInternet();
        }
    }

    private void checkInternet() {
        //WifiManager wifi = (WifiManager) mContext.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(mContext);
        alertDialogBuilder
                .setTitle("Internet Settings")
                .setMessage("No internet. \nCheck the internet connection and try again.")
                .setCancelable(false)
                .setNegativeButton("No", (dialog, id) -> ((Activity) mContext).finish())
                //.setPositiveButton("YES", (dialog, id) -> wifi.setWifiEnabled(true))
                .setPositiveButton("YES", (dialog, id) -> {
                    ((Activity) mContext).finish();
                    Intent splashIntent = new Intent(mContext, SplashActivity.class);
                    mContext.startActivity(splashIntent);
                })
                .create()
                .show();
    }
}

File: app/src/main/java/com/rahmitufanoglu/retrofitdemo/receiver/NetworkChangeReceiver.kt
package com.rahmitufanoglu.retrofitdemo.receiver


import android.app.Activity
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.support.v7.app.AlertDialog
import android.widget.Toast

import com.rahmitufanoglu.retrofitdemo.SplashActivity
import com.rahmitufanoglu.retrofitdemo.util.NetworkUtil


class NetworkChangeReceiver : BroadcastReceiver() {

    private var mContext: Context? = null

    override fun onReceive(context: Context, intent: Intent) {
        mContext = context

        val isConnected = NetworkUtil.isNetworkConnected(context)
        val toast: Toast
        if (isConnected) {
            toast = Toast.makeText(context, "Internet connected", Toast.LENGTH_LONG)
            toast.show()
            mContext!!.startActivity(intent)
            (mContext as Activity).finish()
        } else {
            toast = Toast.makeText(mContext, "Internet connection lost", Toast.LENGTH_LONG)
            toast.show()
            checkInternet()
        }
    }

    private fun checkInternet() {
        // WifiManager wifi = (WifiManager) mContext.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        // var wifi = mContext.getApplicationContext().getSystemService(Context.WIFI_SERVICE) as WifiManager
        val alertDialogBuilder = AlertDialog.Builder(mContext!!)
        alertDialogBuilder
                .setTitle("Internet Settings")
                .setMessage("No internet. \nCheck the internet connection and try again.")
                .setCancelable(false)
                .setNegativeButton("No") { dialog, id -> (mContext as Activity).finish() }
                //.setPositiveButton("YES", (dialog, id) -> wifi.setWifiEnabled(true))
                .setPositiveButton("YES") { dialog, id ->
                    (mContext as Activity).finish()
                    val splashIntent = Intent(mContext, SplashActivity::class.java)
                    mContext!!.startActivity(splashIntent)
                }
                .create()
                .show()
    }

}
